akka-typed-tests/src/test/scala/akka/typed/TimerSpec.scala

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed
+
+import scala.concurrent.duration._
+import scala.util.control.NoStackTrace
+
+import akka.typed.scaladsl.Actor._
+import akka.typed.scaladsl.AskPattern._
+import akka.typed.scaladsl.TimerScheduler
+import akka.typed.testkit.TestKitSettings
+import akka.typed.testkit.scaladsl._
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicInteger
+
+@org.junit.runner.RunWith(classOf[org.scalatest.junit.JUnitRunner])
+class TimerSpec extends TypedSpec("""
+  akka.loglevel = DEBUG
+  """) {
+
+  sealed trait Command
+  case class Tick(n: Int) extends Command
+  case object Bump extends Command
+  case class SlowThenBump(latch: CountDownLatch) extends Command
+  case object End extends Command
+  case class Throw(e: Throwable) extends Command
+  case object Cancel extends Command
+  case class SlowThenThrow(latch: CountDownLatch, e: Throwable) extends Command
+
+  sealed trait Event
+  case class Tock(n: Int) extends Event
+
+  class Exc extends RuntimeException("simulated exc") with NoStackTrace
+
+  trait RealTests extends StartSupport {
+    implicit def system: ActorSystem[TypedSpec.Command]
+    implicit val testSettings = TestKitSettings(system)
+
+    val interval = 1.second
+    val dilatedInterval = interval.dilated
+
+    def target(monitor: ActorRef[Event], timer: TimerScheduler[Command], bumpCount: Int): Behavior[Command] = {
+      def bump(): Behavior[Command] = {
+        val nextCount = bumpCount + 1
+        timer.startTimer("T", Tick(nextCount), interval, repeat = true)
```

Comment:
the API with `repeat = true` looks good in scala however in java it's IMHO a bit confusing to have a boolean in there - always one more guessing step what the boolean means. Sure we don't want to have startPeriodicTimer or other ways to have a dedicated name for them?

Maybe `timer.scheduleOnce` / `timer.schedulePeriodic`? The issue with current scheduler names is that "`schedule`" is periodic, but does not immediately look like it is, so sometimes people do that and don't realise they did a periodic one I thought?